From 5c60679cf1e2325c229677aef04feb4f4ea4d86d Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Mon, 17 Feb 2014 11:44:25 +0800
Subject: [PATCH 3437/5965] m6tvd: remove MESON_M6C_ENHANCEMENT

in kernel 3.10,
	CONFIG_ARCH_MESON6TV  for m6tv  + m6c
	CONFIG_ARCH_MESON6TVD for m6tvd + m6c
---
 arch/arm/mach-meson6tv/Kconfig                |  6 -----
 arch/arm/mach-meson6tvd/Kconfig               |  6 -----
 .../mach-meson6tvd/include/mach/register.h    | 23 -------------------
 drivers/amlogic/display/osd/osd_hw.c          |  1 -
 4 files changed, 36 deletions(-)

diff --git a/arch/arm/mach-meson6tv/Kconfig b/arch/arm/mach-meson6tv/Kconfig
index 7cb634e7edb5..a25c2dca9029 100755
--- a/arch/arm/mach-meson6tv/Kconfig
+++ b/arch/arm/mach-meson6tv/Kconfig
@@ -1,11 +1,5 @@
 if ARCH_MESON6TV
 
-config MESON_M6C_ENHANCEMENT
-       bool "M6C chip enhancement based on M6TV"
-       default n
-       help
-               If using M6C chip instead of M6W or M6L .
-
 config MESON_M6TV_RECOVERY_MODE
 	bool "Recovery mode"
 	help
diff --git a/arch/arm/mach-meson6tvd/Kconfig b/arch/arm/mach-meson6tvd/Kconfig
index 5c0d3a6f1774..bc597a572cdd 100755
--- a/arch/arm/mach-meson6tvd/Kconfig
+++ b/arch/arm/mach-meson6tvd/Kconfig
@@ -1,11 +1,5 @@
 if ARCH_MESON6TVD
 
-config MESON_M6C_ENHANCEMENT
-       bool "M6C chip enhancement based on M6TV"
-       default n
-       help
-               If using M6C chip instead of M6W or M6L .
-
 config MESON_M6TV_RECOVERY_MODE
 	bool "Recovery mode"
 	help
diff --git a/arch/arm/mach-meson6tvd/include/mach/register.h b/arch/arm/mach-meson6tvd/include/mach/register.h
index f49cd5a1abf8..3dd415d8dfce 100755
--- a/arch/arm/mach-meson6tvd/include/mach/register.h
+++ b/arch/arm/mach-meson6tvd/include/mach/register.h
@@ -7464,28 +7464,6 @@
 #define P_MMC_ONE_DATA_CNT 		APB_REG_ADDR(MMC_ONE_DATA_CNT) 	///../ucode/pctl.h
 
 
-#ifndef CONFIG_MESON_M6C_ENHANCEMENT
-#define MMC_CHAN0_CTRL 0x6100
-#define P_MMC_CHAN0_CTRL		APB_REG_ADDR(MMC_CHAN0_CTRL)
-#define MMC_CHAN1_CTRL 0x6104
-#define P_MMC_CHAN1_CTRL		APB_REG_ADDR(MMC_CHAN1_CTRL)
-#define MMC_CHAN2_CTRL 0x6108
-#define P_MMC_CHAN2_CTRL		APB_REG_ADDR(MMC_CHAN2_CTRL)
-#define MMC_CHAN3_CTRL 0x610c
-#define P_MMC_CHAN3_CTRL		APB_REG_ADDR(MMC_CHAN3_CTRL)
-#define MMC_CHAN4_CTRL 0x6110
-#define P_MMC_CHAN4_CTRL		APB_REG_ADDR(MMC_CHAN4_CTRL)
-#define MMC_CHAN5_CTRL 0x6114
-#define P_MMC_CHAN5_CTRL		APB_REG_ADDR(MMC_CHAN5_CTRL)
-#define MMC_CHAN6_CTRL 0x6118
-#define P_MMC_CHAN6_CTRL		APB_REG_ADDR(MMC_CHAN6_CTRL)
-#define MMC_CHAN7_CTRL 0x611c
-#define P_MMC_CHAN7_CTRL		APB_REG_ADDR(MMC_CHAN7_CTRL)
-#define MMC_CHAN8_CTRL 0x6120
-#define P_MMC_CHAN8_CTRL		APB_REG_ADDR(MMC_CHAN8_CTRL)
-#define MMC_CHAN9_CTRL 0x6124
-#define P_MMC_CHAN9_CTRL		APB_REG_ADDR(MMC_CHAN9_CTRL)
-#else
 #define MMC_CHAN_CTRL0 0x6100
 #define P_MMC_CHAN_CTRL0		APB_REG_ADDR(MMC_CHAN_CTRL0)
 #define MMC_CHAN_CTRL1 0x6104
@@ -7499,7 +7477,6 @@
 #define MMC_QOS7_CTRL0 0x609c
 #define P_MMC_QOS7_CTRL0		APB_REG_ADDR(MMC_QOS7_CTRL0)
 
-#endif
 
 #define DC_CAV_CTRL 0x6300 	///../ucode/pctl.h
 #define P_DC_CAV_CTRL 		APB_REG_ADDR(DC_CAV_CTRL) 	///../ucode/pctl.h
diff --git a/drivers/amlogic/display/osd/osd_hw.c b/drivers/amlogic/display/osd/osd_hw.c
index abdbc009316d..2b6bd560f8cc 100755
--- a/drivers/amlogic/display/osd/osd_hw.c
+++ b/drivers/amlogic/display/osd/osd_hw.c
@@ -2348,7 +2348,6 @@ void osd_init_hw(u32  logo_loaded)
 	//here we will init default value ,these value only set once .
 #if defined(CONFIG_ARCH_MESON6TVD)||(defined(CONFIG_ARCH_MESON6TV))
 	aml_set_reg32_mask(P_VPU_OSD1_MMC_CTRL, 1<<12); // set OSD to vdisp2
-	aml_write_reg32(P_MMC_CHAN4_CTRL, 0xc01f); // adjust vdisp weight and age limit
 #endif
 	if(!logo_loaded)
 	{
-- 
2.19.0

